exporter/prometheusexporter/collector.go

```diff
@@ -0,0 +1,537 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusexporter
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type metricValue struct {
+	desc *prometheus.Desc
+
+	labelValues []string
+	metricType  prometheus.ValueType
+	timestamp   time.Time
+	updated     time.Time
+	isHistogram bool
+
+	value float64
+
+	histogramPoints map[float64]uint64
+	histogramSum    float64
+	histogramCount  uint64
```

Comment:
Can you store the pdata points here instead of timestamp, updated, value, histogram*?

Reply:
Maybe this can be added later and done in a follow-up PR.